### PR TITLE
feat: bid auction

### PR DESCRIPTION
**Pull request by topether21**
Enable Create and Accept Bid Auction. Just the functionality. UX/UI is basic just for demonstration. It should be handle in another PR. 

How to test this PR:
0. You require at least two addresses. Please play attention to your fee rate.
1. Login with any Wallet (Alby, Metamask or Unisat).
2. Pick an inscription click on Bid
3. Create a Bid. 
4. Login with the owner address.
5. Accept the Bid.

https://github.com/deezy-inc/deezy-place/assets/126987350/ecfa2444-4fc2-49c4-861a-36766786f626

Fee 8 sats/vB

image

image

### CODE REVIEW

File: src/containers/product-details/index.js [hunk 1]
```diff
@@ -88,75 +99,68 @@ const ProductDetailsArea = memo(
...
-          type: "Created",
-          value: minted,
-        },
-      ];
-    };
-
     const [isOwner, setIsOwner] = useState(false);
 
     useEffect(() => {
-      if (!nostrOrdinalsAddress || !inscription) return;
```
Comment (habibitcoin) [inline src/containers/product-details/index.js:144]: Just curious, what was the change here for?

File: src/containers/product-details/index.js [hunk 2]
```diff
@@ -200,323 +204,341 @@ const ProductDetailsArea = memo(
       auctionNextPriceDrop &&
       auctionNextPriceDrop.scheduledTime > new Date().getTime();
 
-    const isActionsAvailable = () => {
```
Comment (habibitcoin) [inline src/containers/product-details/index.js:203]: I wonder if some of this logic was why some of the boxes disappeared after a bid was made?

File: src/containers/product-details/index.js [hunk 3]
```diff
@@ -200,323 +204,341 @@ const ProductDetailsArea = memo(
...
-
-      return !isOwner && nostr && nostr.value && !isUtxoSpent;
-    };
+    const isWalletConnected = ordinalsPublicKey && nostrOrdinalsAddress;
+    const hasNostrEvent = nostr && nostr.value;
+    const shouldShowSend = isOwner;
+    const shouldShowSell = isOwner && !isUtxoSpent;
+    const shouldShowAuction = isOwner && !isUtxoSpent && !auctionNextPriceDrop;
+    const shouldShowBuyWithLightning =
+      !isOwner && hasNostrEvent && !isUtxoSpent;
```
Comment (habibitcoin) [inline src/containers/product-details/index.js:213]: this evaluates to false for some reason after a bid is placed

File: src/containers/product-details/index.js [hunk 4]
```diff
@@ -200,323 +204,341 @@ const ProductDetailsArea = memo(
...
+    const shouldShowBuyWithBitcoin = hasNostrEvent && !isUtxoSpent;
```
Comment (habibitcoin) [inline src/containers/product-details/index.js:214]: this evaluates to false for some reason after a bid is placed

Comment (habibitcoin) [reply src/containers/product-details/index.js:214]: Even after page refresh

**Review by habibitcoin — CHANGES_REQUESTED** [review]
Amazing work! Was able to make a bid via Alby and accept it via Metamask. The transaction appears to be constructed correctly. The dummies go back to Deezy. I receive my ordinal at 10k sats. Owner receives correct bid of 25000 sats. I net-spent 38844 sats, 8,844 was fees (21.41 sat/vByte, I selected 26). 25,000 was for the bid. And 5k was to top up my ordinal to 10k output. Excellent. 

image

Also tested bidding w/ Metamask, accepting via Alby. Worked perfect. Except the fee rate was too low again at 14.54. Final tx was exact same size. Hmm

Few small things:
- When a bid is placed, all other info disappears related to buying or dutch auction (if one is ongoing)
image
- Want to 2x check the fee math. The final tx ended up being 614b, 414vB, and the final fee rate was slightly under what I expected.
- @dannydeezy question on the acceptance UX, I'm a little worried that we should have some sort of dialogue explaining what the "Check" does. Maybe a follow up
image


I don't have funds in my Unisat wallet or Xverse so could not test those yet.

**Review by habibitcoin — CHANGES_REQUESTED** [review]
IGNORE THIS REVIEW**

**Review by habibitcoin — APPROVED** [review]
Nice work!! Lookin good.

Given the large size of the PR, I'd like @andredevjs to also take a look before merge

**Comment by dannydeezy** [comment]
nit: should say "Bid on* this Ordinal"
Screenshot 2023-08-10 at 5 38 30 PM

**Comment by dannydeezy** [comment]
@habibitcoin I made a bid on my baseball card!

https://deploy-preview-298.test.deezy.place/inscription/b312d2ac6ed1fee29acb6be9d1cf07e001595a6e9a1a26fef324552da290e343i0

**Comment by dannydeezy** [comment]
should i be able to view bids on an item I don't own? cant' remember what we decided here. i think it would be nice eventually, but not a blocker
Screenshot 2023-08-10 at 5 40 45 PM

**Comment by habibitcoin** [comment]
> should i be able to view bids on an item I don't own? cant' remember what we decided here. i think it would be nice eventually, but not a blocker Screenshot 2023-08-10 at 5 40 45 PM

Not yet, we're going to have that as a follow up once this is merged: https://linear.app/deezy/issue/ENG-137/create-a-view-to-see-other-competing-bids

And ayyy, the tx looks for your bid! Hope you don't mind tho I'm gonna keep this piece of cave art haha bullish on Deezy

**Comment by dannydeezy** [comment]
The "View All Bids" button is very slow to load.
I'm ok with optimizing this later, but I think we should have a temporary button or placeholder that say "Searching for Bids..." with either a loading icon or the "..."'s changing just to convey that we are looking for bids.

https://github.com/deezy-inc/deezy-place/assets/43550949/ceb1d508-5dce-458c-b76e-dadf3d6d479c

**Comment by dannydeezy** [comment]
It's not clear what "Address" is in this column. I think that's the "Buyer" in which case it should say "Buyer". Otherwise i think it might be my receive address

Screenshot 2023-08-11 at 10 23 27 AM

**Comment by dannydeezy** [comment]
I made a bid with Unisat at 24 sat/vbyte and was able to successfully accept the bid via Alby. Broadcasted at 23.9 sat/vbyte, very nice!

**Comment by topether21** [comment]
@dannydeezy What about this?
https://github.com/deezy-inc/deezy-place/assets/126987350/d1547b59-a25e-4575-8cdf-de93f69ae2a4

**Comment by topether21** [comment]
@dannydeezy 
image

**Pull request merged**